tools/cellprofiler/cellprofiler.xml

```diff
@@ -0,0 +1,94 @@
+<tool id="cp319" name="Cellprofiler" version="@CP_VERSION@">
+    <description>Run a CellProfiler pipeline</description>
+    
+    <macros>
+        <import>macros.xml</import>
+        <xml name="test_assert_content" token_n="291">
+            <assert_contents>
+                <has_n_lines n="@N@" />
+            </assert_contents>
+        </xml>
+    </macros>
+
+    <expand macro="requirements" version="@CP_VERSION@" package="cellprofiler" />
+    <expand macro="stdio" />
+
+    <command><![CDATA[
+        bash '$script_file' &&
+        cellprofiler -c -r --file-list list.txt -o ./output -p '$pipeline' 
+    ]]></command>
+
+    <configfiles>
+        <configfile name="script_file">
+mkdir ./input
+mkdir ./output
+#for $i,$filename in enumerate($inputdir):
+    ln -s '$filename' ./input/${filename.element_identifier}.${filename.ext}
+#end for
+
+find `pwd`/input -name "*.*" > list.txt
+
+        </configfile>
+    </configfiles>
+  
+    <inputs>
+        <param name="pipeline" type="data" format="txt" label="Pipeline file"/>
```

Comment:
please add a help, what is this pipeline file ... you can create this with various CP modules ....